efficientdet/det_model_fn.py

```diff
@@ -156,24 +173,26 @@ def focal_loss(y_pred, y_true, alpha, gamma, label_smoothing, normalizer):
     alpha: A float32 scalar multiplying alpha to the loss from positive examples
       and (1-alpha) to the loss from negative examples.
     gamma: A float32 scalar modulating loss from hard and easy examples.
+    normalizer: Divide loss by this value.
     label_smoothing: Float in [0, 1]. If > `0` then smooth the labels.
-    normalizer: Divide loss by this value
+
+  Returns:
+    loss: A float32 scalar representing normalized total loss.
   """
   with tf.name_scope('focal_loss'):
     alpha = tf.convert_to_tensor(alpha, dtype=y_pred.dtype)
     gamma = tf.convert_to_tensor(gamma, dtype=y_pred.dtype)
 
-    # Get the cross_entropy for each entry
-    ce = tf.keras.losses.categorical_crossentropy(
-        y_true, y_pred, from_logits=True, label_smoothing=label_smoothing)
+    # apply label smoothing.
+    y_true = y_true * (1.0 - label_smoothing) + 0.5 * label_smoothing
 
-    pred_prob = tf.sigmoid(y_pred)
+    # get cross_entropy for each entry.
+    ce = tf.nn.sigmoid_cross_entropy_with_logits(labels=y_true, logits=y_pred)
```

Comment:
tf.keras.losses.categorical_crossentropy will change the output shape, so I have to use tf.nn.sigmoid_cross_entropy_with_logits  here.

Reply:
:+1: